godot-macros/src/godot_api.rs

```diff
@@ -247,43 +256,52 @@ fn transform_trait_impl(original_impl: Impl) -> Result<TokenStream, Error> {
                 godot_init_impl = quote! {
                     impl ::godot::obj::cap::GodotInit for #class_name {
                         fn __godot_init(base: ::godot::obj::Base<Self::Base>) -> Self {
-                            <Self as ::godot::bind::GodotExt>::init(base)
+                            <Self as #trait_name>::init(base)
                         }
                     }
                 };
                 create_fn = quote! { Some(#prv::callbacks::create::<#class_name>) };
             }
 
             "to_string" => {
+                to_string_impl = quote! {
+                    impl ::godot::obj::cap::GodotToString for #class_name {
+                        fn __virtual_to_string(&self) -> ::godot::builtin::GodotString {
+                            <Self as #trait_name>::to_string(self)
+                        }
+                    }
+                };
+
                 to_string_fn = quote! { Some(#prv::callbacks::to_string::<#class_name>) };
             }
 
             // Other virtual methods, like ready, process etc.
-            known_name if VIRTUAL_METHOD_NAMES.contains(&known_name) => {
+            _ => {
                 let method = util::reduce_to_signature(method);
 
                 // Godot-facing name begins with underscore
-                virtual_method_names.push(format!("_{method_name}"));
+                let virtual_method_name = {
+                    if method_name == "init_ext" {
+                        String::from("_init")
+                    } else {
+                        format!("_{method_name}")
+                    }
+                };
```

Comment:
Maybe add a quick comment referencing the `_init` -> `init_ext` transformation in codegen, otherwise this looks like magic here.

Also, the outer `{ }` block is confusing; you can directly use `let virtual_method_name = if ...`.

Reply:
:+1: